RuntimeException when a `SavedState` recovery is triggered at `EditVariationAttributesFragment`

Summary
==========
The `EditVariationAttributesFragment` is failing to recover itself during any kind of `SavedState` restoration from the ViewModel, throwing the following exception:

```
2021-05-06 02:24:14.736 23636-23636/com.woocommerce.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.woocommerce.android, PID: 23636
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android/com.woocommerce.android.ui.main.MainActivity}: java.lang.IllegalArgumentException: No destination with ID 2131362794 is on the NavController's back stack. The current destination is null
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5456)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5362)
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.IllegalArgumentException: No destination with ID 2131362794 is on the NavController's back stack. The current destination is null
        at androidx.navigation.NavController.getBackStackEntry(NavController.java:1346)
        at com.woocommerce.android.ui.products.variations.attributes.edit.EditVariationAttributesModule$Companion.provideSavedStateRegistryOwner(EditVariationAttributesModule.kt:29)
        at com.woocommerce.android.ui.products.variations.attributes.edit.EditVariationAttributesModule.provideSavedStateRegistryOwner(Unknown Source:2)
        at com.woocommerce.android.ui.products.variations.attributes.edit.EditVariationAttributesModule_ProvideSavedStateRegistryOwnerFactory.provideSavedStateRegistryOwner(EditVariationAttributesModule_ProvideSavedStateRegistryOwnerFactory.java:35)
        at com.woocommerce.android.DaggerWooCommerceDebug_HiltComponents_SingletonC$MainActivitySubcomponentImpl$EditVariationAttributesFragmentSubcomponentImpl.savedStateRegistryOwner(DaggerWooCommerceDebug_HiltComponents_SingletonC.java:4808)
        at com.woocommerce.android.DaggerWooCommerceDebug_HiltComponents_SingletonC$MainActivitySubcomponentImpl$EditVariationAttributesFragmentSubcomponentImpl.viewModelFactory(DaggerWooCommerceDebug_HiltComponents_SingletonC.java:4816)
        at com.woocommerce.android.DaggerWooCommerceDebug_HiltComponents_SingletonC$MainActivitySubcomponentImpl$EditVariationAttributesFragmentSubcomponentImpl.injectEditVariationAttributesFragment(DaggerWooCommerceDebug_HiltComponents_SingletonC.java:4827)
        at com.woocommerce.android.DaggerWooCommerceDebug_HiltComponents_SingletonC$MainActivitySubcomponentImpl$EditVariationAttributesFragmentSubcomponentImpl.inject(DaggerWooCommerceDebug_HiltComponents_SingletonC.java:4821)
        at com.woocommerce.android.DaggerWooCommerceDebug_HiltComponents_SingletonC$MainActivitySubcomponentImpl$EditVariationAttributesFragmentSubcomponentImpl.inject(DaggerWooCommerceDebug_HiltComponents_SingletonC.java:4768)
        at dagger.android.DispatchingAndroidInjector.maybeInject(DispatchingAndroidInjector.java:113)
        at dagger.android.DispatchingAndroidInjector.inject(DispatchingAndroidInjector.java:134)
        at dagger.android.support.AndroidSupportInjection.inject(AndroidSupportInjection.java:75)
        at dagger.android.support.AndroidSupportInjection.inject(AndroidSupportInjection.java:67)
        at com.woocommerce.android.ui.base.BaseDaggerFragment.onAttach(BaseDaggerFragment.kt:22)
        at androidx.fragment.app.Fragment.performAttach(Fragment.java:2922)
        at androidx.fragment.app.FragmentStateManager.attach(FragmentStateManager.java:464)
2021-05-06 02:24:14.738 23636-23636/com.woocommerce.android E/AndroidRuntime:     at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:275)
        at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:112)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3126)
        at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:3059)
        at androidx.fragment.app.Fragment.restoreChildFragmentState(Fragment.java:1891)
        at androidx.fragment.app.Fragment.onCreate(Fragment.java:1867)
        at androidx.navigation.fragment.NavHostFragment.onCreate(NavHostFragment.java:264)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
        at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
        at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:112)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3126)
        at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:3059)
        at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:240)
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:276)
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:115)
        at com.woocommerce.android.ui.main.AppUpgradeActivity.onCreate(AppUpgradeActivity.kt:58)
        at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:185)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        	... 13 more
2021-05-06 02:24:14.738 23636-23636/com.woocommerce.android I/Sentry: Uncaught exception received.
2021-05-06 02:24:14.740 23636-23636/com.woocommerce.android D/Sentry: Capturing event: 7f68b24bba0148d8935be04aa891768d
2021-05-06 02:24:14.771 23636-23636/com.woocommerce.android D/Sentry: Event was dropped by beforeSend
2021-05-06 02:24:14.774 23636-23786/com.woocommerce.android D/Sentry: Adding Envelope to offline storage: /data/user/0/com.woocommerce.android/cache/sentry/ec92c949-60f9-4d5a-a937-c2470d1484f0.envelope
2021-05-06 02:24:14.776 23636-23786/com.woocommerce.android D/Sentry: Disk flush envelope fired
2021-05-06 02:24:14.776 23636-23636/com.woocommerce.android I/Sentry: Invoking inner uncaught exception handler.
2021-05-06 02:24:14.795 23636-23636/com.woocommerce.android I/Process: Sending signal. PID: 23636 SIG: 9

```

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesFragment.kt
@@ -2,7 +2,7 @@ package com.woocommerce.android.ui.products.variations.attributes.edit
 
 import android.os.Bundle
 import android.view.View
-import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
 import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
 import androidx.navigation.fragment.navArgs
@@ -31,7 +31,7 @@ class EditVariationAttributesFragment :
 
     @Inject lateinit var viewModelFactory: ViewModelFactory
 
-    private val viewModel: EditVariationAttributesViewModel by activityViewModels { viewModelFactory }
+    private val viewModel: EditVariationAttributesViewModel by viewModels { viewModelFactory }
 
     private val navArgs: EditVariationAttributesFragmentArgs by navArgs()
 

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesModule.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesModule.kt
@@ -2,9 +2,7 @@ package com.woocommerce.android.ui.products.variations.attributes.edit
 
 import android.os.Bundle
 import androidx.lifecycle.ViewModel
-import androidx.navigation.fragment.findNavController
 import androidx.savedstate.SavedStateRegistryOwner
-import com.woocommerce.android.R
 import com.woocommerce.android.di.ViewModelAssistedFactory
 import com.woocommerce.android.ui.products.variations.attributes.edit.EditVariationAttributesViewModel.Factory
 import com.woocommerce.android.viewmodel.ViewModelKey
@@ -22,16 +20,13 @@ abstract class EditVariationAttributesModule {
         fun provideDefaultArgs(fragment: EditVariationAttributesFragment): Bundle? {
             return fragment.arguments
         }
-
-        @JvmStatic
-        @Provides
-        fun provideSavedStateRegistryOwner(fragment: EditVariationAttributesFragment): SavedStateRegistryOwner {
-            return fragment.findNavController().getBackStackEntry(R.id.nav_graph_products)
-        }
     }
 
     @Binds
     @IntoMap
     @ViewModelKey(EditVariationAttributesViewModel::class)
     abstract fun bindFactory(factory: Factory): ViewModelAssistedFactory<out ViewModel>
+
+    @Binds
+    abstract fun bindSavedStateRegistryOwner(fragment: EditVariationAttributesFragment): SavedStateRegistryOwner
 }

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesViewModel.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/edit/EditVariationAttributesViewModel.kt
@@ -12,7 +12,6 @@ import com.woocommerce.android.util.CoroutineDispatchers
 import com.woocommerce.android.viewmodel.LiveDataDelegate
 import com.woocommerce.android.viewmodel.MultiLiveEvent.Event.Exit
 import com.woocommerce.android.viewmodel.MultiLiveEvent.Event.ExitWithResult
-import com.woocommerce.android.viewmodel.ResourceProvider
 import com.woocommerce.android.viewmodel.SavedStateWithArgs
 import com.woocommerce.android.viewmodel.ScopedViewModel
 import dagger.assisted.Assisted
@@ -26,8 +25,7 @@ class EditVariationAttributesViewModel @AssistedInject constructor(
     @Assisted savedState: SavedStateWithArgs,
     dispatchers: CoroutineDispatchers,
     private val productRepository: ProductDetailRepository,
-    private val variationRepository: VariationDetailRepository,
-    private val resources: ResourceProvider
+    private val variationRepository: VariationDetailRepository
 ) : ScopedViewModel(savedState, dispatchers) {
     private val _editableVariationAttributeList =
         MutableLiveData<List<VariationAttributeSelectionGroup>>()

